playwright/advanced-search-page/advancedSearch.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { test } from "@playwright/test";
+import { AdvancedSearchPage } from "./advanced-search-page";
+
+test.describe("Advanced Search", () => {
+  test("Navigate to the advanced search page and all page elements are displayed", async ({
+    page,
+  }) => {
+    const advancedSearchPage = new AdvancedSearchPage(page);
+    await advancedSearchPage.navigateToAdvancedSearchPage();
+    await advancedSearchPage.clickAdvancedSearchLink();
+    await advancedSearchPage.verifyElementVisible(
+      advancedSearchPage.advancedSearchHeading
+    );
+    await advancedSearchPage.verifyElementVisible(
+      advancedSearchPage.keywordSearchLabel
+    );
+    await advancedSearchPage.verifyElementVisible(
+      advancedSearchPage.keywordSearchBox
+    );
+    await advancedSearchPage.verifyElementVisible(
+      advancedSearchPage.authorSearchLabel
+    );
+    await advancedSearchPage.verifyElementVisible(
+      advancedSearchPage.authorSearchBox
+    );
+    await advancedSearchPage.verifyElementVisible(
+      advancedSearchPage.subjectSearchLabel
+    );
+    await advancedSearchPage.verifyElementVisible(
+      advancedSearchPage.subjectSearchBox
+    );
+    await advancedSearchPage.verifyElementVisible(
+      advancedSearchPage.titleSearchLabel
+    );
+    await advancedSearchPage.verifyElementVisible(
+      advancedSearchPage.titleSearchBox
+    );
+    await advancedSearchPage.verifyElementVisible(
+      advancedSearchPage.dateFilterFromLabel
+    );
+    await advancedSearchPage.verifyElementVisible(
+      advancedSearchPage.dateFilterFromField
+    );
+    await advancedSearchPage.verifyElementVisible(
+      advancedSearchPage.dateFilterToLabel
+    );
+    await advancedSearchPage.verifyElementVisible(
+      advancedSearchPage.dateFilterToField
+    );
+    await advancedSearchPage.verifyElementVisible(
+      advancedSearchPage.advancedSearchButton
+    );
+    await advancedSearchPage.verifyElementVisible(
+      advancedSearchPage.advancedSearchClearButton
+    );
```

Comment:
What are your thoughts on simplifying this and iterating over the props of the page object and checking if all the Locator props are visible?

Reply:
There currently is no easy way in TypeScript to check if a type matches a custom interface, i.e. Locator.

Reply:
ugh